lib/root_controller.js

```diff
@@ -0,0 +1,174 @@
+      // (Terminal freezes imediatelly after poweroff the Tessel)
+
+      ch.once('error', function(e) {
+        if (e === 255) {
+          process.stdout.write('\u001b[2J\u001b[0;0H');
+          logs.warn('Sorry, you are not authorized!');
+          logs.info('"t2 key generate" might help :-)');
+          reject();
+        } else {
+          logs.warn('Error while connected to ' + tessels[id].connections[0].host + ':', e);
+          reject(e);
+        }
+
+      });
+      ch.once('exit', function(e) {
+
+        if (e === 255) {
+          process.stdout.write('\u001b[2J\u001b[0;0H');
+          logs.warn('Sorry, you are not authorized!');
+          logs.info('"t2 key generate" might help :-)');
+          reject();
+        } else if (e === 127) { // exit by user
+          // clear console
+          process.stdout.write('\u001b[2J\u001b[0;0H');
+          logs.warn('Connection to ' + tessels[id].connections[0].host + ' closed!\n');
+          resolve();
+        } else if (e === 0) {
+          // everything works fine ... now lets clear the terminal
+          process.stdout.write('\u001b[2J\u001b[0;0H');
+          logs.info('Connection to ' + tessels[id].connections[0].host + ' closed!\n');
+          resolve();
+        } else {
+          logs.warn('Connection to ' + tessels[id].connections[0].host + ' closed due to reason:\n', e);
+          reject(e);
+        }
+      });
+
+      ch.once('close', function(e) {
+        // tessel powers off
+        if (e === 0) {
+          // everything works fine...
+          resolve();
+        } else if (e === 255) {
+          //reject();
+          process.exit();
+        } else if (e === 127) {
+          reject();
+        } else {
+          logs.warn('Connection to ' + tessels[id].connections[0].host + ' closed due to reason:\n', e);
+          reject();
+        }
+      });
+    }
+  },
+  pro: function(rtm, resolve, reject) {
+    // Menu navigation
+
+    // raw mode for gain ability to navigate up and down within the menu
+    process.stdin.setRawMode(true);
+
+    process.stdin.pipe(rtm.createStream()).pipe(process.stdout);
+
+    rtm.on('close', function() {
+      process.stdin.setRawMode(false);
+      process.stdin.end();
+    });
+    rtm.on('error', function(e) {
+      process.stdin.setRawMode(false);
+      reject(e);
+    });
+  },
+  clear: function(resolve) {
+    // selected exit!
+    process.stdout.write('\u001b[2J\u001b[0;0H');
+    controller.ssh.exit(resolve);
+  },
+  exit: function(resolve) {
+    resolve();
+    //process.exit();
+  },
+  multipleTessels: function(opts, tessels, rtm, resolve, reject) {
+    // Hack for creating usebility (exit menu)
+    tessels.push({
+      connection: {
+        connectionType: 'exit'
+      }
+    });
+
+    rtm.reset();
+    rtm.write('                                                \n');
+    // create menu entries
+    for (var i in tessels) {
```

Comment:
@rwaldron I need to add the reference to rtm so I replaced the former foreach by for in